packages/@uppy/provider-views/src/utils/PartialTreeUtils/getBreadcrumbs.ts

```diff
@@ -0,0 +1,31 @@
+import type {
+  PartialTree,
+  PartialTreeFolder,
+  PartialTreeFolderNode,
+  PartialTreeId,
+} from '@uppy/core/lib/Uppy'
+
+const getBreadcrumbs = (
+  partialTree: PartialTree,
+  currentFolderId: PartialTreeId,
+): PartialTreeFolder[] => {
+  let folder = partialTree.find(
+    (f) => f.id === currentFolderId,
+  ) as PartialTreeFolder
```

Comment:
```suggestion
  )!
```

Reply:
If we only do `!`, typescript wouldn't know if `folder` is a `PartialTreeFolder` or `PartialTreeFile` - and it needs to know that `folder` is a `PartialTreeFolder`, because that's what `getBreadcrumbs()` must return.

Coercion here is unavoidable, because the fact that `currentFolderId` is a folder id stems from our UI, Typescript can't know about this.